File: 이수창/20230520/체육복.js
function solution(n, lost, reserve) {
    let clothes = Array.from({length:n},()=>1)
    //clothes=[1,1,1,1,1]
    var answer = 0;
    for(let i=0;i<clothes.length;i++)
    {
        if(lost.filter((a)=>{return(a-1)==i}).length)
         clothes[i]--   //1번쟤 요소의 개수를 1 감소 clothes=[1,0,1,0,1]
        if(reserve.filter((a)=>{return(a-1)==i}).length)
         clothes[i]++  //clothes[2,0,2,0,2]
    }
    console.log(clothes)
    for(let j=0;j<clothes.length;j++)
    {  

        if(clothes[j]==0&&clothes[j-1]>1)
        {
            clothes[j]++
            clothes[j-1]--
            continue;
        }
        if(clothes[j]==0&&clothes[j+1]>1)
        {
            clothes[j]++
            clothes[j+1]--
            continue;
        }

    }
    answer=clothes.filter((a)=>{return(a>0)}).length

    return answer;
}

File: 서예진B/20230520/체육복.py
def solution(n, lost, reserve):
    answer = 0
    # 학생별 보유 체육복 수를 담은 배열 res
    res = [1 for _ in range(n)]
    # 여벌 체육복 + 1
    for idx in reserve:
        res[idx-1] += 1
    # 도난 체육복 - 1
    for idx in lost:
        res[idx-1] -= 1
    
    # 체육복 수를 순회하며
    for i in range(n):
        # 체육복이 없다면
        if res[i] == 0:
            # 양 옆 번호 학생들 체육복 수 조사
            # 첫번째 학생이면 두번째 학생만 확인
            if i == 0:
                if res[i+1] > 1:
                    res[i+1] -= 1
                    res[i] = 1
            # 양쪽 학생 모두 확인
            elif 1 <= i < n-1:
                if res[i-1] > 1:
                    res[i-1] -= 1
                    res[i] = 1
                elif res[i+1] > 1:
                    res[i+1] -= 1
                    res[i] = 1
            # 마지막 학생이면 바로 앞 학생만 확인
            elif i == n-1:
                if res[i-1] > 1:
                    res[i-1] -= 1
                    res[i] = 1
    # 체육복 수를 순회하며        
    for cnt in res:
        # 체육복이 0보다 크면 수업에 참여 가능
        if cnt > 0:
            answer += 1
            
    return answer